### ISSUE TITLE
Type of literal char is not the literal char itself

### ISSUE THREAD

**Issue by jsmaniac** [racket/typed-racket#434]
### What version of Racket are you using?

6.6.0.5
### What program did you run?

``` racket
#lang typed/racket
(ann #\nul #\nul)
(ann #\a #\a)
(ann #\b '#\b)
```
### What should have happened?

The type of a char `#\c` should be `#\c`.
### If you got an error message, please include it here.

```
Type Checker: type mismatch
  expected: #\nul
  given: Char in: #\nul
```

**Comment by AlexKnauth** [comment]
I haven't tested it, but it seems like this code here:
https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/typecheck/tc-literal.rkt#L23
Could be changed from
```racket
(~and i (~or :number :str :bytes))
```
To
```racket
(~and i (~or :number :str :bytes :char))
```
That would allow the first case, the `[i:exp expected]` to detect it.

**Comment by samth** [comment]
That seems right to me too.

Sam

**Comment by jsmaniac** [comment]
Closed by #527 .
